### PR TITLE
use subtest for table units (pkg/scheduler)

### PR DESCRIPTION
**Pull request by xchapter7x**
**What this PR does / why we need it**: Update scheduler's unit table tests to use subtest

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
breaks up PR: https://github.com/kubernetes/kubernetes/pull/63281
/ref #63267

**Release note**:

```release-note
This PR will leverage subtests on the existing table tests for the scheduler units.
Some refactoring of error/status messages and functions to align with new approach.

```

### CODE REVIEW

File: pkg/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -177,79 +183,113 @@ func TestScheduler(t *testing.T) {
...
-		}
-		if e, a := item.expectErrorPod, gotPod; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: error pod: wanted %v, got %v", i, e, a)
-		}
-		if e, a := item.expectForgetPod, gotForgetPod; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: forget pod: wanted %v, got %v", i, e, a)
-		}
-		if e, a := item.expectError, gotError; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: error: wanted %v, got %v", i, e, a)
+	s, _ := NewFromConfigurator(configurator, nil...)
```
Comment (misterikkit) [inline pkg/scheduler/scheduler_test.go]: Another weird one. `nil...` is the same as not passing any args after configurator. (don't fix in this PR. I'm taking notes though.)

File: pkg/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -675,15 +742,17 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
...
 		expectError        error
 		expectPodBind      *v1.Binding
 		expectAssumeCalled bool
 		expectBindCalled   bool
 		eventReason        string
 		volumeBinderConfig *persistentvolume.FakeVolumeBinderConfig
 	}{
-		"all-bound": {
+		{
+			name: "all-bound",
```
Comment (misterikkit) [inline pkg/scheduler/scheduler_test.go]: It's okay to use spaces in test names. t.Run will replace them with underscore.

File: pkg/scheduler/scheduler_test.go [hunk 3]
```diff
@@ -694,7 +763,8 @@ func TestSchedulerWithVolumeBinding(t *testing.T) {
 
 			eventReason: "Scheduled",
 		},
-		"bound,invalid-pv-affinity": {
+		{
+			name: "bound,invalid-pv-affinity",
```
Comment (misterikkit) [inline pkg/scheduler/scheduler_test.go]: The commas could probably be replaced with slashes in these test names. (I'm assuming you already know how `-run` interacts with subtests and subsubtests.)

**Comment by xchapter7x** [comment]
/assign @misterikkit 
this PR should be in a good spot if you have a moment to take a look. thanks.

File: pkg/scheduler/scheduler_test.go [hunk 4]
```diff
@@ -177,79 +183,113 @@ func TestScheduler(t *testing.T) {
...
+				eventBroadcaster,
+				testNode,
+				t,
+			)
+		})
+	}
+}
+
+func testScheduler(
+	injectBindError error,
```
Comment (misterikkit) [inline pkg/scheduler/scheduler_test.go]: As we discussed in https://github.com/kubernetes/kubernetes/pull/63665#issuecomment-393884049, let's keep this inline for now.

File: pkg/scheduler/scheduler_test.go [hunk 5]
```diff
@@ -381,53 +421,80 @@ func TestSchedulerErrorWithLongBinding(t *testing.T) {
 	conflictPod := podWithPort("bar", "", 8080)
 	pods := map[string]*v1.Pod{firstPod.Name: firstPod, conflictPod.Name: conflictPod}
 	for _, test := range []struct {
+		Name            string
```
Comment (misterikkit) [inline pkg/scheduler/scheduler_test.go]: lowercase name

**Review by bsalamat — APPROVED** [review]
/approve

Thanks, @xchapter7x!

**Pull request merged**